### PR TITLE
adopt -> adoptium, Java 16 -> Java 17

### PR DESCRIPTION
**Pull request by rossabaker**

### CODE REVIEW

**Comment by rossabaker** [comment]
This is stuck on the way we silence errors in tests.  We need a replacement for `AccessController`.

**Comment by rossabaker** [comment]
@vasilmkd Do you know what's going on with the failing builds here?  Looks like it's still looking for an `adopt`.

**Comment by rossabaker** [comment]
Oh... I see it now.

**Comment by vasilmkd** [comment]
As far as I can see, you managed to resolve it, right?

**Comment by rossabaker** [comment]
Yeah, my grep failed me.  Sorry for the ping.

File: testing/src/test/scala/org/http4s/testing/ErrorReporting.scala [hunk 1]
```diff
@@ -35,28 +27,12 @@ object ErrorReporting {
...
-      val domains = new Array[ProtectionDomain](1)
-      domains(0) = domain
-      val context = new AccessControlContext(domains)
-
       // Redirect output to dummy stream
       val fakeOutStream = new PrintStream(NullOutStream)
       val fakeErrStream = new PrintStream(NullOutStream)
       System.setOut(fakeOutStream)
       System.setErr(fakeErrStream)
-      try java.security.AccessController.doPrivileged(
```
Comment (rossabaker) [inline testing/src/test/scala/org/http4s/testing/ErrorReporting.scala:55]: I don't recall why this was ever necessary.  Tests pass without it, and it's only used in tests.

**Comment by vasilmkd** [comment]
No worries, I wish I could be around http4s some more, I'll be back soon.

**Review by armanbilge — APPROVED** [review]

**Pull request merged**